exercises/hangman/Example.cs

```diff
@@ -1,85 +1,91 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Linq;
-
-public delegate void HangmanChangedEventHandler(object sender, HangmanState state);
-
+using System.Reactive;
+using System.Reactive.Subjects;
+
+/// <summary>
+/// Returned state is <see href="https://blog.ndepend.com/c-sharp-immutable-types-understanding-attraction/">immutable</see>,
+/// to help the students implement the right solution, escaping the trap of <see cref="RemainingGuesses">RemainingGuesses</see> be the same on
+/// all observed elements.
+/// </summary>
 public class HangmanState
 {
-    public HangmanStatus Status { get; set; }
-    public int RemainingGuesses { get; set; }
-    public string MaskedWord { get; set; }
-    public HashSet<char> Guesses { get; set; }
+    public string MaskedWord { get; }
+    public ImmutableHashSet<char> GuessedChars { get; }
+    public int RemainingGuesses { get; }
+
+    public HangmanState(string maskedWord, ImmutableHashSet<char> guessedChars, int remainingGuesses)
+    {
+        MaskedWord = maskedWord;
+        GuessedChars = guessedChars;
+        RemainingGuesses = remainingGuesses;
+    }
 }
 
-public enum HangmanStatus
+/// <summary>
```

Comment:
I think this comment can be removed, as I expect almost to know the rules of hangman and for those that don't, there is a link in the README.

Maybe `GameLostException` would be a slightly more better name, or perhaps `TooManyGuessesException`? The latter is the most descriptive I think.